src/cc/libbpf.c

```diff
@@ -177,9 +187,14 @@ int bpf_prog_load(enum bpf_prog_type prog_type,
   if (ret < 0 && !log_buf) {
     // caller did not specify log_buf but failure should be printed,
     // so call recursively and print the result to stderr
+    buffer_size = attr.insn_cnt * 128;
```

Comment:
> Also if we need to implement this for a c api, who owns the memory in this case?

I think it's a good time to fix up this API a little bit. Right now either the callee or bpf_prog_load itself can allocate memory, which is why the `insn_cnt * 128` got duplicated in multiple places. If instead someone needs to allocate an increasing buffer, the exponential increase logic would get duplicated as well. I would like to avoid that.

I see a couple options, would appreciate folks thoughts:

1. Always allocate memory inside bpf_prog_load, and if the caller passed in a pointer, assign the buffer to that pointer and let the caller print+free the memory, otherwise print+free locally. This has the downside that the memory allocation escapes the function, making different language bindings slightly more complex.

2. Remove the log_buf parameter, and just always print to stderr. This breaks python's ability to print a customized string, and could break other callers as well.

3. Pass in a file handle to print to instead of a buffer. This might also not work well for all language bindings.